contracts/PriceOracle/PriceOracle.sol

```diff
@@ -0,0 +1,166 @@
+
+        // Number of decimals determines whether the price needs to be multiplied or divided for scaling
+        // Handle the 2 scenarios separately to ensure a non-fractional scale value
+        if (decimals <= 36) {
+            // Decimals is always >=0 so the scale max value is 1e36 here and not at risk of overflowing
+            uint256 scale = 10 ** (36 - decimals);
+            return FullMath.mulDiv(price, scale, config.baseUnit);
+        } else {
+            // Decimals is capped at 72 by earlier validation so scale max value is 1e36 here and not at risk of overflowing
+            uint256 scale = 10 ** (decimals - 36);
+            // Divide price by scale and base unit separately
+            // Multiplying scale and base unit before dividing price could result in an overflow
+            uint256 scaledPrice = FullMath.mulDiv(price, 1, scale);
+            return FullMath.mulDiv(scaledPrice, 1, config.baseUnit);
+        }
+    }
+
+    /**
+     * @notice Retrieves the token config for a particular cToken address
+     * @param cToken The cToken address that the token config should be returned for
+     */
+    function getConfig(address cToken) external view returns (TokenConfig memory) {
+        TokenConfig memory config = tokenConfigs[cToken];
+        // Check if config exists for cToken
+        require(config.cToken != address(0), "Config for cToken does not exist");
+        return config;
+    }
+
+    /**
+     * @notice Adds a new token config to enable the contract to provide prices for a new asset
+     * @param config Token config struct that contains the info for a new asset configuration
+     */
+    function addConfig(TokenConfig calldata config) external onlyOwner {
+        // Check if duplicate configs were submitted for the same cToken
+        require(tokenConfigs[config.cToken].cToken == address(0), "Config for cToken already exists");
+        validateTokenConfig(config);
+
+        tokenConfigs[config.cToken] = config;
+        emit PriceOracleAssetAdded(config.cToken, config.baseUnit, config.priceFeed);
+    }
+
+    /**
+     * @notice Updates the price feed in the token config for a particular cToken
+     * @param cToken The cToken address that the config needs to be updated for
+     * @param priceFeed The address of the new price feed the config needs to be updated to
+     */
+    function updateConfigPriceFeed(address cToken, address priceFeed) external onlyOwner {
+        TokenConfig storage config = tokenConfigs[cToken];
+        // Check if config exists for cToken
+        require(config.cToken != address(0), "Config does not exist for cToken");
+        // Validate price feed
+        require(priceFeed != address(0), "Invalid new price feed address");
+        require(config.priceFeed != priceFeed, "Price feed address same as the existing one");
+
+        emit PriceOracleAssetPriceFeedUpdated(cToken, config.priceFeed, priceFeed);
```

Comment:
We generally like to emit events after state changes.

Reply:
I'll get this updated but was just trying to avoid setting an extra variable to maintain the old parameter. But out of curiosity, this is just a styling preference right? Since the whole method would execute or revert so there's no risk of a disconnect between the emit and the state change?